src/GitLink/Linker.cs

```diff
@@ -23,234 +26,81 @@ namespace GitLink
                 }
 
-                if(!srcSrvContext.RawUrl.Contains("%var2%") && !srcSrvContext.RawUrl.Contains("{0}"))
+                string commitId;
+                commitId = repository.Head.Commits.First().Sha;
+
+                string rawUrl = provider.RawGitUrl;
+                if (!rawUrl.Contains("%var2%") && !rawUrl.Contains("{0}"))
                 {
-                    srcSrvContext.RawUrl = string.Format("{0}/{{0}}/%var2%", srcSrvContext.RawUrl);
+                    rawUrl = string.Format("{0}/{{0}}/%var2%", rawUrl);
                 }
-				
-                foreach (var compilable in compilables)
-                {
-                    var relativePathForUrl = compilable.Replace(context.SolutionDirectory, string.Empty).Replace("\\", "/");
-                    while (relativePathForUrl.StartsWith("/"))
-                    {
-                        relativePathForUrl = relativePathForUrl.Substring(1, relativePathForUrl.Length - 1);
-                    }
 
-                    srcSrvContext.Paths.Add(new Tuple<string, string>(compilable, relativePathForUrl));
+                var srcSrvContext = new SrcSrvContext
+                {
+                    RawUrl = rawUrl,
+                    DownloadWithPowershell = options.DownloadWithPowerShell,
+                    Revision = commitId,
+                };
+                foreach (string sourceFile in sourceFiles)
+                {
+                    string repoRelativePath = Catel.IO.Path.GetRelativePath(sourceFile, Path.GetDirectoryName(repositoryDirectory))
+                        .Replace('\\', '/');
+                    srcSrvContext.Paths.Add(Tuple.Create(sourceFile, repoRelativePath));
                 }
 
-                // When using the VisualStudioTeamServicesProvider, add extra infomration to dictionary with VSTS-specific data
-                if (context.Provider.GetType().Name.EqualsIgnoreCase("VisualStudioTeamServicesProvider"))
+                if (provider is Providers.VisualStudioTeamServicesProvider)
                 {
-                    srcSrvContext.VstsData["TFS_COLLECTION"] = context.Provider.CompanyUrl;
-                    srcSrvContext.VstsData["TFS_TEAM_PROJECT"] = context.Provider.ProjectName;
-                    srcSrvContext.VstsData["TFS_REPO"] = context.Provider.ProjectName;
+                    srcSrvContext.VstsData["TFS_COLLECTION"] = provider.CompanyUrl;
+                    srcSrvContext.VstsData["TFS_TEAM_PROJECT"] = provider.ProjectName;
+                    srcSrvContext.VstsData["TFS_REPO"] = provider.ProjectName;
                 }
 
-                project.CreateSrcSrv(projectSrcSrvFile, srcSrvContext);
-
-                Log.Debug("Created source server link file, updating pdb file '{0}'", context.GetRelativePath(projectPdbFile));
-
-                PdbStrHelper.Execute(pdbStrFile, projectPdbFile, projectSrcSrvFile);
+                ProjectExtensions.CreateSrcSrv(projectSrcSrvFile, srcSrvContext);
```

Comment:
We will need to refactor the ProjectExtensions (we don't want to actually use ProjectExtensions as a caller stub here).